### PR TITLE
order quick inserter items by frecency

### LINKED ISSUE #23866: Quick Inserter: Most recent not correctly showing recent blocks
The new quick inserter should show the 6 most recent blocks when opened, in addition to the search field. This appears the case in an install that adds no big block collections.

default

However regardless of how many times I insert the Table block, it never appears in the list of most recent.

There's a different issue also, though — as soon as I activate the Jetpack plugin, all the blocks in the quick inserter are taken over:

jetpack

These also stay firmly in place, regardless of usage. The same is the case with Ultimate Gutenberg Addons:

ultimate

Note that Jetpack is using `registerBlockCollection` in case that matters.

### PR DESCRIPTION
**Pull request by youknowriad**
Closes #23866 

**Testing instructions**

- Make sure the quick inserter items are ordered by frecency

### CODE REVIEW

File: packages/block-editor/src/components/inserter/quick-inserter.js [hunk 1]
```diff
@@ -46,7 +51,11 @@ function QuickInserterList( {
 	onHover,
 } ) {
 	const shownBlockTypes = useMemo(
-		() => blockTypes.slice( 0, SHOWN_BLOCK_TYPES ),
+		() =>
+			orderBy( blockTypes, [ 'frecency' ], [ 'desc' ] ).slice(
```
Comment (ntsekouras) [inline packages/block-editor/src/components/inserter/quick-inserter.js:55]: Should we implement this inside `searchBlockItems`? It seems business logic related to the search function. What do you think?

Comment (youknowriad) [reply packages/block-editor/src/components/inserter/quick-inserter.js:55]: No, I don't think so, in fact, it was initially in the selector (so applied to everything) but due to design decisions that are per component, we moved it out, and now there are places where we want things to be sorted and others not.

Comment (ntsekouras) [reply packages/block-editor/src/components/inserter/quick-inserter.js:55]: Yeah, I meant as a new param in searchBlockItems.

Comment (youknowriad) [reply packages/block-editor/src/components/inserter/quick-inserter.js:55]: It's not always tied to search. For instance if the search input is empty, this list is rendered still without calling `searchBlockItems`

**Review by ntsekouras — APPROVED** [review]
Thanks @youknowriad!

**Pull request merged**